From: PF4Public@users.noreply.github.com
Subject: LIN Fix miscellaneous build failures

First one fixes following build error:

FAILED: v8_context_snapshot_generator 
"python3.11" "../../build/toolchain/gcc_link_wrapper.py" --output="./v8_context_snapshot_generator" -- x86_64-pc-linux-gnu-clang++ -fuse-ld=lld -Wl,--build-id=sha1 -fPIC -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now -Wl,--icf=all -Wl,--color-diagnostics -Wl,--undefined-version -no-canonical-prefixes -rdynamic -Wl,-z,defs -Wl,--as-needed -pie -Wl,--disable-new-dtags -Wl,--icf=none -Wl,-O1 -Wl,--as-needed -Wl,--sort-common -Wl,-z,norelro -march=haswell -O3 -pipe -falign-functions=32 -fno-unwind-tables -fno-asynchronous-unwind-tables -fno-stack-protector -fomit-frame-pointer -fno-semantic-interposition -fno-plt -o "./v8_context_snapshot_generator" -Wl,--start-group @"./v8_context_snapshot_generator.rsp"  -Wl,--end-group  -latomic -ldl -lpthread -lrt -lgmodule-2.0 -lglib-2.0 -lgobject-2.0 -lgthread-2.0 -ljsoncpp -licui18n -licuuc -licudata -lsmime3 -lnss3 -lnssutil3 -lplds4 -lplc4 -lnspr4 -lz -levent -lresolv -lgio-2.0 -ljpeg -lpng16 -lxml2 -lxslt -lexpat -lwebpdemux -lwebpmux -lwebp -lfreetype -lfontconfig -lharfbuzz-subset -lharfbuzz -lopus -lavcodec -lavformat -lavutil -lopenh264 -lX11 -lXcomposite -lXdamage -lXext -lXfixes -lXrender -lXrandr -lXtst -ldrm -lxcb -lxkbcommon -lgbm -lXi -lpci -latk-1.0 -latk-bridge-2.0 -lpangocairo-1.0 -lpango-1.0 -lcairo -lasound -lFLAC -lsnappy -ldbus-1 -latspi 
ld.lld: error: undefined symbol: content_settings::WebsiteSettingsRegistry::GetInstance()
>>> referenced by content_settings.cc
>>>               common/content_settings.o:(RendererContentSettingRules::IsRendererContentSetting(ContentSettingsType)) in archive obj/components/content_settings/core/common/libcommon.a

ld.lld: error: undefined symbol: content_settings::WebsiteSettingsRegistry::begin() const
>>> referenced by content_settings.cc
>>>               common/content_settings.o:(RendererContentSettingRules::IsRendererContentSetting(ContentSettingsType)) in archive obj/components/content_settings/core/common/libcommon.a

ld.lld: error: undefined symbol: content_settings::WebsiteSettingsRegistry::end() const
>>> referenced by content_settings.cc
>>>               common/content_settings.o:(RendererContentSettingRules::IsRendererContentSetting(ContentSettingsType)) in archive obj/components/content_settings/core/common/libcommon.a
x86_64-pc-linux-gnu-clang++: error: linker command failed with exit code 1 (use -v to see invocation)

Second one fixes following build error:

In file included from ../../chrome/browser/browser_process.cc:5: 
In file included from ../../chrome/browser/browser_process.h:24: 
In file included from ../../components/component_updater/adblock_updater_service.h:33: 
In file included from ../../components/component_updater/download_filters_task.h:29:
../../services/network/public/cpp/shared_url_loader_factory.h:13:10: fatal error: 'services/network/public/mojom/url_loader_factory.mojom.h' file not found 
#include "services/network/public/mojom/url_loader_factory.mojom.h" 
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 1 error generated.

---
--- a/tools/v8_context_snapshot/BUILD.gn
+++ b/tools/v8_context_snapshot/BUILD.gn
@@ -87,6 +87,7 @@
         "//services/service_manager/public/cpp",
         "//third_party/blink/public:blink",
         "//v8",
+        "//components/content_settings/core/browser",
       ]
 
       configs += [
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -134,6 +134,7 @@
     "//build:chromeos_buildflags",
     "//chrome/common:buildflags",
     "//media:media_buildflags",
+    "//services/network/public/mojom",
   ]
 }
 
